packages/cursorless-engine/src/processTargets/modifiers/surroundingPair/index.ts

```diff
@@ -148,3 +160,22 @@ function processSurroundingPairCore(
     scopeType,
   );
 }
+
+function nodeHasError(node: SyntaxNode, includeChildren = false): boolean {
```

Comment:
This seems quite conservative. Are there test cases that fail if you don't eg check grand-siblings?

Reply:
The added test for example. The error node is a child to an ancestor of the selected node.

Reply:
as discussed, let's use your proposed version for now and if it becomes bothersome we can revisit.  see also #1600